src/Data/EqP.hs

```diff
@@ -88,6 +94,17 @@ instance (EqP a, EqP b) => EqP (a :*: b) where
 instance EqP TR.TypeRep where
     eqp x y = TR.SomeTypeRep x == TR.SomeTypeRep y
 
+#if MIN_VERSION_base(4,18,0)
+instance EqP TL.SChar where
+    eqp _ _ = True
```

Comment:
This is wrong, isn't it. (Look at above `TypeRep` instance). `EqP` and `OrdP` allow heterogenous comparison:

```haskell
class (forall a. Eq (f a)) => EqP (f :: k -> Type) where
    eqp :: f a -> f b -> Bool
          -- ^ ---- ^ --
```

Reply:
My apologies, I misunderstood what `EqP` is doing. When I get a chance, I'll change this implementation to something like:

```hs
instance EqP SChar where
  eqp x y = fromSChar x == fromSChar y
```

Reply:
I've updated all of the `EqP` and `OrdP` instances to use the intended semantics.